File: tales_of_a_sort/tales.py
def main():
    case_num = int(input())
    for _ in range(case_num):
        n = int(input())
        nums = input().split()
        nums = [eval(i) for i in nums]
        _max = 0
        num_moves = 0
        for j in range(n - 1):
            if nums[j] > nums[j + 1] and nums[j] > _max:
                num_moves = nums[j]
                _max = num_moves
        print(num_moves)
    
    
if __name__ == "__main__":
    main()

File: tales_of_a_sort/tales.c
#include <stdio.h>
#include <stdlib.h>

int main()
{
    int case_num;
    scanf("%d", &case_num);
    for (int i = 0; i < case_num; i++)
    {
        int n;
        scanf("%d", &n);
        int *nums = malloc(sizeof(int) * n);
        int max = 0;
        for (int j = 0; j < n; j++)
        {
            scanf("%d", nums + j);
        }

        int num_moves = 0;
        for (int j = 0; j < n - 1; j++)
        {
            if (nums[j] > nums[j + 1] && nums[j] > max)
            {
            
                num_moves = nums[j];
                max = num_moves;
            }
        }

        printf("%d\n", num_moves);
    }
}
